### DISCUSSION TITLE
Add support in Selection Tool's No Antialiasing option for deformation, rotation, scaling and deletion

### DISCUSSION THREAD

**Discussion by 2dvision** [opentoonz/opentoonz#6732] [General]
There is no difference when delete select area with "No Antialiasing" enable or disable.

![selection tool -no antialiasing- dont work](https://cloud.githubusercontent.com/assets/16509592/16408667/1adbb7c6-3d2f-11e6-84f7-a1c3dea0b4f4.gif)

--- 
Status
---
- [x] Known Issue
- [ ] Documented Workaround(s)
- [ ] Documented Code Location(s)
- [ ] Priority Established
- [ ] Bounty Attached
- [ ] Developer Assigned

----

**Comment by artisteacher** [comment]
I’m seeing the same thing with both raster and toonz raster levels. Looking with preview, there looks to be antialiasing either way.

**Comment by RodneyBaker** [comment]
Something strange is going on here but it's not entirely clear what that is.
I note that turning "No Antialiasing' on or off in Toonz Premium also does not appear to do anything.

Perhaps there is another usage for this we aren't taking into consideration?

**Comment by RodneyBaker** [comment]
Not that this particularly helps but for completeness sake...

From the Opentoonz documentation:

> No Antialiasing when activated, the antialiasing is not applied when the selection is deformed or rotated. This is available on Toonz Raster and Raster drawings only.

REF:  https://opentoonz.readthedocs.io/en/latest/drawing_animation_levels.html#using-the-selection-tool

The option does appear to make a difference when rotating images drawn on toonz raster levels.
Having the option off tends to produce better results although YMMV.

Recommend closing this as working as designed (although not optimally!) with an eye for looking to enhance the feature's processing in the future.

**Comment by gab3d** [comment]
...and please let me note that this behavior (being able to transform without AA and preserving the original drawing, as much as possible) is essential to preserve pixel-art content while transforming it.

As per what I've just tested, currently this functionality leaves a lot to be desired, both when rotating and when scaling. 😞 

here a couple of video screen captures showing the (bad) results:
[transfSelection.zip](https://github.com/opentoonz/opentoonz/files/4011442/transfSelection.zip)

**Comment by RodneyBaker** [comment]
Part of this surely relates to the general antialiasing throughput that Opentoonz uses by default.
It's much better than it was a year or so ago but yes... still leaves a lot to be desired.

I was recently working using a workflow I don't normally use and was amazed to see sharp pixels being maintained from start to finish in the process.  Now, if I can only remember what settings I was using when I did that!

Added:  Some of this is coming back to me.
I was testing Shun Iwasawa's new feature that allows changing Level Settings across multiple cells at the same time.  There were three main points of interest;  1) toggle Antialiasing on but leave smoothness set to 0.  2) Draw/paint with a brush that produces crisp lines (or use the Pencil) 3) Some other thing I've forgotten.  :(   

The key however was to start with crisp pixels, process crisp pixels and don't add anything that will mess up that process (unless doing so for a reason when going final).

I confirm this works quite well although I cannot attest to the actual process going on in the code itself.

**Comment by RodneyBaker** [comment]
Untagging as 'bug' as this appears to be working as designed although... needs additional design work.
Aliasing/Antialiasing is a process it would be good to trace from ingest to output to optimize the process, reduce variation, and streamline workflow especially given the number of users who have expressed interest.

**Comment by gab3d** [comment]
What is really strange in the case of Rotation, is that the preview of the rotation is perfect quality, but when accepting the operation, OT seems to process it someway and it goes completely berserk. If it just preserved the preview as the final result, it'd be great.

For Scale the preview is already bad, so it'd take a little rethinking, I guess.

**Comment by RodneyBaker** [comment]
There are several factors here that point to other issues and it's not entirely clear how related they are.

For instance, simply drawing a rectangle with Geometry tool on raster/toonz raster (with line thickness of 1)... in Pencil Mode... should result in a rectangle of 1 unit line thickness but instead results in just the corners being drawn.  This as if the aliasing calculation is negatively impacting the results.  There is a report opened on that:  #2887 .

This makes me think that perhaps an integer is being used instead of a float or rounding is occurring resulting in data loss.  Something similar might be happening in this case also.  Or in the case of a desired aliasing/stair stepping of pixels the data that should be removed is left intact.

**Comment by gab3d** [comment]
yes, I see that here too.
...even when choosing pixels as unit of measure; which has no sense to me.

**Comment by RodneyBaker** [comment]
@2dvision  I've adjusted the title of this report and added the feature request label as the manual explicitly states two limitations of the current tool as not working with deformation and rotation.  Cutting/deletion isn't specifically mentioned in that and should be but beyond all that... the option can certainly use an update to allow for the desired results.  This report has gone a long way in better understanding what the current feature does do and does not.

**Comment by RodneyBaker** [comment]
Updated title to refocus on what is desired to be added to the current feature set of 'No Antialiasing' option in the Selection Tool.

To restate:
The 'No Antialiasing' option is not currently designed to support deformations, rotations, scaling and deletion.   There is a very brief (and incomplete) mention of this in the Opentoonz documentation.  
REF:  https://opentoonz.readthedocs.io/en/latest/drawing_animation_levels.html?highlight=antialiasing#using-the-selection-tool

Also:  This relates tangentially to #1782.
